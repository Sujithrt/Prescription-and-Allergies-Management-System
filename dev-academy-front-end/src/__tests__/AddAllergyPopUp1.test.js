import React from 'react';
import { shallowToJson } from "enzyme-to-json";
import { shallow, mount } from 'enzyme';
import AddAllergyPopUp from '../components/addAllergyPopUp/AddAllergyPopUp'
import axios from "axios"
import MockAdapter from "axios-mock-adapter"
import { patient, allergy } from '../testConstants'
import { Router } from 'react-router-dom'

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useLocation: () => ({
        pathname: "localhost:3000/patient/1001"
    })
}));

describe("AddAllergyPopUp Test", () => {
    it("AddAllergyPopUp is rendered as expected", () => {
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />);
        expect(shallowToJson(addAllergyPopUp)).toMatchSnapshot();
    });
});

describe("User input Test", () => {
    it("Allergy Name change with user input", () => {
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />);
        addAllergyPopUp.find({ 'data-testid': "allergyName" }).simulate('change', { target: { 'data-testid': 'allergyName', value: 'Paracetamol Allergy' } })
        expect(addAllergyPopUp.find({ 'data-testid': "allergyName" }).prop('value')).toEqual('Paracetamol Allergy');
    });
    it("Allergy Reaction change with user input", () => {
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />);
        addAllergyPopUp.find({ 'data-testid': "allergyReaction" }).simulate('change', { target: { 'data-testid': 'allergyReaction', value: 'Tiredness' } })
        expect(addAllergyPopUp.find({ 'data-testid': "allergyReaction" }).prop('value')).toEqual('Tiredness');
    });
    it("Allergy Ingredients change with user input", () => {
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />);
        const ingredient = { label: "Paracetamol", value: 1 }
        addAllergyPopUp.find({ 'data-testid': "ingredients" }).simulate('change', { target: { 'data-testid': 'ingredients', value: ingredient } })
        expect(addAllergyPopUp.find({ 'data-testid': "ingredients" }).prop('value').target.value).toEqual(ingredient);
    });
    it("Allergy Severity change with user input", () => {
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />);
        const severity = { label: "MILD", value: "MILD" }
        addAllergyPopUp.find({ 'data-testid': "severity" }).simulate('change', { target: { 'data-testid': 'severity', value: severity } })
        expect(addAllergyPopUp.find({ 'data-testid': "severity" }).prop('value').target.value).toEqual(severity);
    });
    it("Allergy Status change with user input", () => {
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />);
        const status = { label: "ACTIVE", value: "ACTIVE" }
        addAllergyPopUp.find({ 'data-testid': "status" }).simulate('change', { target: { 'data-testid': 'status', value: status } })
        expect(addAllergyPopUp.find({ 'data-testid': "status" }).prop('value').target.value).toEqual(status);
    });
});

describe('test add allergy in new encounter', () => {
    let mock;

    beforeAll(() => {
        mock = new MockAdapter(axios);
        jest.mock("react-router-dom", () => ({
            ...jest.requireActual("react-router-dom"),
            useLocation: () => ({
                search: null
            })
        }));
    });

    afterEach(() => {
        mock.reset();
    });
    it('add allergy with all details filled', () => {
        mock.onPost().reply(201, allergy)
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />);
        addAllergyPopUp.find({ 'data-testid': "allergyName" }).simulate('change', { target: { 'data-testid': 'allergyName', value: 'Paracetamol Allergy' } })
        addAllergyPopUp.find({ 'data-testid': "allergyReaction" }).simulate('change', { target: { 'data-testid': 'allergyReaction', value: 'Tiredness' } })
        addAllergyPopUp.find({ 'data-testid': "ingredients" }).simulate('change', { target: { 'data-testid': 'ingredients', value: { label: "Paracetamol", value: 1 } } })
        addAllergyPopUp.find({ 'data-testid': "severity" }).simulate('change', { target: { 'data-testid': 'severity', value: { label: "MILD", value: "MILD" } } })
        addAllergyPopUp.find({ 'data-testid': "status" }).simulate('change', { target: { 'data-testid': 'status', value: { label: "ACTIVE", value: "ACTIVE" } } })
        addAllergyPopUp.find({ 'data-testid': "addAllergyButton" }).simulate('click', { preventDefault: () => { } })
    });
    it('add allergy with some details filled', () => {
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />);
        addAllergyPopUp.find({ 'data-testid': "allergyName" }).simulate('change', { target: { 'data-testid': 'allergyName', value: '' } })
        addAllergyPopUp.find({ 'data-testid': "allergyReaction" }).simulate('change', { target: { 'data-testid': 'allergyReaction', value: 'Tiredness' } })
        addAllergyPopUp.find({ 'data-testid': "ingredients" }).simulate('change', { target: { 'data-testid': 'ingredients', value: { label: "Paracetamol", value: 1 } } })
        addAllergyPopUp.find({ 'data-testid': "severity" }).simulate('change', { target: { 'data-testid': 'severity', value: { label: "MILD", value: "MILD" } } })
        addAllergyPopUp.find({ 'data-testid': "status" }).simulate('change', { target: { 'data-testid': 'status', value: { label: "ACTIVE", value: "ACTIVE" } } })
        addAllergyPopUp.find({ 'data-testid': "addAllergyButton" }).simulate('click', { preventDefault: () => { } })
    });
    it('Allergy already exists error', () => {
        mock.onPost().reply(409)
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />);
        addAllergyPopUp.find({ 'data-testid': "allergyName" }).simulate('change', { target: { 'data-testid': 'allergyName', value: 'Paracetamol Allergy' } })
        addAllergyPopUp.find({ 'data-testid': "allergyReaction" }).simulate('change', { target: { 'data-testid': 'allergyReaction', value: 'Tiredness' } })
        addAllergyPopUp.find({ 'data-testid': "ingredients" }).simulate('change', { target: { 'data-testid': 'ingredients', value: { label: "Paracetamol", value: 1 } } })
        addAllergyPopUp.find({ 'data-testid': "severity" }).simulate('change', { target: { 'data-testid': 'severity', value: { label: "MILD", value: "MILD" } } })
        addAllergyPopUp.find({ 'data-testid': "status" }).simulate('change', { target: { 'data-testid': 'status', value: { label: "ACTIVE", value: "ACTIVE" } } })
        addAllergyPopUp.find({ 'data-testid': "addAllergyButton" }).simulate('click', { preventDefault: () => { } })
    });
    it('Unknown error', () => {
        mock.onPost().reply(500)
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />);
        addAllergyPopUp.find({ 'data-testid': "allergyName" }).simulate('change', { target: { 'data-testid': 'allergyName', value: 'Paracetamol Allergy' } })
        addAllergyPopUp.find({ 'data-testid': "allergyReaction" }).simulate('change', { target: { 'data-testid': 'allergyReaction', value: 'Tiredness' } })
        addAllergyPopUp.find({ 'data-testid': "ingredients" }).simulate('change', { target: { 'data-testid': 'ingredients', value: { label: "Paracetamol", value: 1 } } })
        addAllergyPopUp.find({ 'data-testid': "severity" }).simulate('change', { target: { 'data-testid': 'severity', value: { label: "MILD", value: "MILD" } } })
        addAllergyPopUp.find({ 'data-testid': "status" }).simulate('change', { target: { 'data-testid': 'status', value: { label: "ACTIVE", value: "ACTIVE" } } })
        addAllergyPopUp.find({ 'data-testid': "addAllergyButton" }).simulate('click', { preventDefault: () => { } })
    });
});

describe('modal window', () => {
    it('Check open modal window upon add allergy button click', () => {
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />)
        addAllergyPopUp.find({ 'data-testid': 'openAddAllergyModal' }).simulate('click')
    });
    it('Check close modal window upon x button click', () => {
        const addAllergyPopUp = shallow(<AddAllergyPopUp patientDetails={patient} />)
        addAllergyPopUp.find({ 'data-testid': 'addAllergyModal' }).simulate('hide')
    });
});

describe('Modal Window clear form', () => {
    it('Check clearAllergyDetails function upon component mount', () => {
        const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
        const addAllergyPopUp = mount(<Router history={historyMock}><AddAllergyPopUp patientDetails={patient} /></Router>)
        expect(shallowToJson(addAllergyPopUp)).toMatchSnapshot()
    });
});
