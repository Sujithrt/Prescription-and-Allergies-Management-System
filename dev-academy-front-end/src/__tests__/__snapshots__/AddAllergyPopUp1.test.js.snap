// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddAllergyPopUp Test AddAllergyPopUp is rendered as expected 1`] = `
<Fragment>
  <Button
    active={false}
    data-testid="openAddAllergyModal"
    disabled={false}
    onClick={[Function]}
    variant="primary"
  >
    Add Allergy
  </Button>
  <Modal
    animation={true}
    aria-labelledby="contained-modal-title-vcenter"
    autoFocus={true}
    backdrop={true}
    centered={true}
    data-testid="addAllergyModal"
    dialogAs={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      }
    }
    enforceFocus={true}
    keyboard={true}
    onHide={[Function]}
    restoreFocus={true}
    show={false}
    size="md"
    style={
      Object {
        "zIndex": 2000,
      }
    }
  >
    <ModalHeader
      closeButton={true}
      closeLabel="Close"
    >
      <ModalTitle
        id="contained-modal-title-vcenter"
      >
        Add Allergy
      </ModalTitle>
    </ModalHeader>
    <ModalBody>
      <form
        onSubmit={[Function]}
      >
        <Container
          fluid={false}
        >
          <Row
            className="justify-content-md-center"
            noGutters={false}
          >
            <Col
              md={10}
            />
          </Row>
          <Row
            className="justify-content-md-center textfield"
            noGutters={false}
          >
            <Col
              md={10}
            >
              <ForwardRef(TextField)
                className="updateInput"
                data-testid="allergyName"
                label="Allergy Name"
                onChange={[Function]}
                value=""
                variant="standard"
              />
            </Col>
          </Row>
          <Row
            className="justify-content-md-center"
            noGutters={false}
          >
            <Col
              md={10}
            >
              <ForwardRef
                className="search-box"
                components={
                  Object {
                    "ClearIndicator": [Function],
                    "Control": [Function],
                    "CrossIcon": [Function],
                    "DownChevron": [Function],
                    "DropdownIndicator": [Function],
                    "Group": [Function],
                    "GroupHeading": [Function],
                    "IndicatorSeparator": [Function],
                    "IndicatorsContainer": [Function],
                    "Input": [Function],
                    "LoadingIndicator": [Function],
                    "LoadingMessage": [Function],
                    "Menu": [Function],
                    "MenuList": [Function],
                    "MenuPortal": [Function],
                    "MultiValue": [Function],
                    "MultiValueContainer": [Function],
                    "MultiValueLabel": [Function],
                    "MultiValueRemove": [Function],
                    "NoOptionsMessage": [Function],
                    "Option": [Function],
                    "Placeholder": [Function],
                    "SelectContainer": [Function],
                    "SingleValue": [Function],
                    "ValueContainer": [Function],
                  }
                }
                data-testid="ingredients"
                isClearable={true}
                loadOptions={[Function]}
                onChange={[Function]}
                placeholder="Ingredient causing allergy"
                required={true}
                value=""
              />
            </Col>
          </Row>
          <Row
            className="justify-content-md-center textfield"
            noGutters={false}
          >
            <Col
              md={10}
            >
              <ForwardRef(TextField)
                className="updateInput"
                data-testid="allergyReaction"
                label="Reaction"
                onChange={[Function]}
                value=""
                variant="standard"
              />
            </Col>
          </Row>
          <Row
            className="justify-content-md-center"
            noGutters={false}
          >
            <Col
              md={10}
            >
              <ForwardRef
                className="search-box"
                components={
                  Object {
                    "ClearIndicator": [Function],
                    "Control": [Function],
                    "CrossIcon": [Function],
                    "DownChevron": [Function],
                    "DropdownIndicator": [Function],
                    "Group": [Function],
                    "GroupHeading": [Function],
                    "IndicatorSeparator": [Function],
                    "IndicatorsContainer": [Function],
                    "Input": [Function],
                    "LoadingIndicator": [Function],
                    "LoadingMessage": [Function],
                    "Menu": [Function],
                    "MenuList": [Function],
                    "MenuPortal": [Function],
                    "MultiValue": [Function],
                    "MultiValueContainer": [Function],
                    "MultiValueLabel": [Function],
                    "MultiValueRemove": [Function],
                    "NoOptionsMessage": [Function],
                    "Option": [Function],
                    "Placeholder": [Function],
                    "SelectContainer": [Function],
                    "SingleValue": [Function],
                    "ValueContainer": [Function],
                  }
                }
                data-testid="severity"
                isClearable={true}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "MILD",
                      "value": "MILD",
                    },
                    Object {
                      "label": "MODERATE",
                      "value": "MODERATE",
                    },
                    Object {
                      "label": "CRITICAL",
                      "value": "CRITICAL",
                    },
                  ]
                }
                placeholder="Allergy Severity"
                required={true}
                value=""
              />
            </Col>
          </Row>
          <Row
            className="justify-content-md-center"
            noGutters={false}
          >
            <Col
              md={10}
            >
              <ForwardRef
                className="search-box"
                components={
                  Object {
                    "ClearIndicator": [Function],
                    "Control": [Function],
                    "CrossIcon": [Function],
                    "DownChevron": [Function],
                    "DropdownIndicator": [Function],
                    "Group": [Function],
                    "GroupHeading": [Function],
                    "IndicatorSeparator": [Function],
                    "IndicatorsContainer": [Function],
                    "Input": [Function],
                    "LoadingIndicator": [Function],
                    "LoadingMessage": [Function],
                    "Menu": [Function],
                    "MenuList": [Function],
                    "MenuPortal": [Function],
                    "MultiValue": [Function],
                    "MultiValueContainer": [Function],
                    "MultiValueLabel": [Function],
                    "MultiValueRemove": [Function],
                    "NoOptionsMessage": [Function],
                    "Option": [Function],
                    "Placeholder": [Function],
                    "SelectContainer": [Function],
                    "SingleValue": [Function],
                    "ValueContainer": [Function],
                  }
                }
                data-testid="status"
                isClearable={true}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "label": "ACTIVE",
                      "value": "ACTIVE",
                    },
                    Object {
                      "label": "INACTIVE",
                      "value": "INACTIVE",
                    },
                  ]
                }
                placeholder="Allergy Status"
                required={true}
                value=""
              />
            </Col>
          </Row>
          <Row
            className="justify-content-md-center"
            noGutters={false}
          >
            <Col
              md={5}
            >
              <Button
                active={false}
                className="updateInput"
                data-testid="addAllergyButton"
                disabled={false}
                onClick={[Function]}
                variant="primary"
              >
                Add
              </Button>
            </Col>
          </Row>
        </Container>
      </form>
    </ModalBody>
  </Modal>
</Fragment>
`;

exports[`Modal Window clear form Check clearAllergyDetails function upon component mount 1`] = `
<Router
  history={
    Object {
      "listen": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
      "location": Object {},
      "push": [MockFunction],
    }
  }
>
  <AddAllergyPopUp
    patientDetails={
      Object {
        "age": 21,
        "allergies": Array [
          Object {
            "encounter": Object {
              "date": "2021-10-26",
              "id": 10,
              "patient": Object {
                "age": 21,
                "city": "Bangalore",
                "email": "leanne.graham@gmail.com",
                "firstName": "Leanne",
                "gender": "Male",
                "id": 1001,
                "lastName": "Graham",
                "medicalConditions": Array [
                  Object {
                    "id": 1,
                    "name": "Cephalalgia (headache)",
                  },
                  Object {
                    "id": 3,
                    "name": "Otitis externa (swimmer's ear)",
                  },
                  Object {
                    "id": 9,
                    "name": "Contusion (bruise)",
                  },
                ],
                "phoneNumber": 9123456780,
                "state": "Karnataka",
              },
              "user": Object {
                "accountNonExpired": true,
                "accountNonLocked": true,
                "authorities": Array [
                  Object {
                    "authority": "USER",
                    "id": 1,
                    "roleCode": "USER",
                    "roleDescription": "User role",
                  },
                  Object {
                    "authority": "ADMIN",
                    "id": 2,
                    "roleCode": "ADMIN",
                    "roleDescription": "Admin role",
                  },
                ],
                "credentialsNonExpired": true,
                "email": "john.cage@gmail.com",
                "enabled": true,
                "id": 1,
                "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                "role": "Physician",
                "username": "John",
              },
            },
            "id": 7,
            "ingredient": Object {
              "id": 4,
              "name": "Aspirin",
            },
            "name": "Aspirin Allergy",
            "patient": Object {
              "age": 21,
              "city": "Bangalore",
              "email": "leanne.graham@gmail.com",
              "firstName": "Leanne",
              "gender": "Male",
              "id": 1001,
              "lastName": "Graham",
              "medicalConditions": Array [
                Object {
                  "id": 1,
                  "name": "Cephalalgia (headache)",
                },
                Object {
                  "id": 3,
                  "name": "Otitis externa (swimmer's ear)",
                },
                Object {
                  "id": 9,
                  "name": "Contusion (bruise)",
                },
              ],
              "phoneNumber": 9123456780,
              "state": "Karnataka",
            },
            "reaction": "Tiredness",
            "severity": "MODERATE",
            "status": "ACTIVE",
          },
          Object {
            "encounter": Object {
              "date": "2021-10-26",
              "id": 13,
              "patient": Object {
                "age": 21,
                "city": "Bangalore",
                "email": "leanne.graham@gmail.com",
                "firstName": "Leanne",
                "gender": "Male",
                "id": 1001,
                "lastName": "Graham",
                "medicalConditions": Array [
                  Object {
                    "id": 1,
                    "name": "Cephalalgia (headache)",
                  },
                  Object {
                    "id": 3,
                    "name": "Otitis externa (swimmer's ear)",
                  },
                  Object {
                    "id": 9,
                    "name": "Contusion (bruise)",
                  },
                ],
                "phoneNumber": 9123456780,
                "state": "Karnataka",
              },
              "user": Object {
                "accountNonExpired": true,
                "accountNonLocked": true,
                "authorities": Array [
                  Object {
                    "authority": "USER",
                    "id": 1,
                    "roleCode": "USER",
                    "roleDescription": "User role",
                  },
                  Object {
                    "authority": "ADMIN",
                    "id": 2,
                    "roleCode": "ADMIN",
                    "roleDescription": "Admin role",
                  },
                ],
                "credentialsNonExpired": true,
                "email": "john.cage@gmail.com",
                "enabled": true,
                "id": 1,
                "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                "role": "Physician",
                "username": "John",
              },
            },
            "id": 8,
            "ingredient": Object {
              "id": 1,
              "name": "Paracetamol",
            },
            "name": "Paracetamol",
            "patient": Object {
              "age": 21,
              "city": "Bangalore",
              "email": "leanne.graham@gmail.com",
              "firstName": "Leanne",
              "gender": "Male",
              "id": 1001,
              "lastName": "Graham",
              "medicalConditions": Array [
                Object {
                  "id": 1,
                  "name": "Cephalalgia (headache)",
                },
                Object {
                  "id": 3,
                  "name": "Otitis externa (swimmer's ear)",
                },
                Object {
                  "id": 9,
                  "name": "Contusion (bruise)",
                },
              ],
              "phoneNumber": 9123456780,
              "state": "Karnataka",
            },
            "reaction": "sdfd",
            "severity": "MILD",
            "status": "ACTIVE",
          },
          Object {
            "encounter": Object {
              "date": "2021-10-29",
              "id": 20,
              "patient": Object {
                "age": 21,
                "city": "Bangalore",
                "email": "leanne.graham@gmail.com",
                "firstName": "Leanne",
                "gender": "Male",
                "id": 1001,
                "lastName": "Graham",
                "medicalConditions": Array [
                  Object {
                    "id": 1,
                    "name": "Cephalalgia (headache)",
                  },
                  Object {
                    "id": 3,
                    "name": "Otitis externa (swimmer's ear)",
                  },
                  Object {
                    "id": 9,
                    "name": "Contusion (bruise)",
                  },
                ],
                "phoneNumber": 9123456780,
                "state": "Karnataka",
              },
              "user": Object {
                "accountNonExpired": true,
                "accountNonLocked": true,
                "authorities": Array [
                  Object {
                    "authority": "USER",
                    "id": 1,
                    "roleCode": "USER",
                    "roleDescription": "User role",
                  },
                  Object {
                    "authority": "ADMIN",
                    "id": 2,
                    "roleCode": "ADMIN",
                    "roleDescription": "Admin role",
                  },
                ],
                "credentialsNonExpired": true,
                "email": "john.cage@gmail.com",
                "enabled": true,
                "id": 1,
                "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                "role": "Physician",
                "username": "John",
              },
            },
            "id": 13,
            "ingredient": Object {
              "id": 3,
              "name": "Ibuprofen",
            },
            "name": "Ibuprofin Allergy",
            "patient": Object {
              "age": 21,
              "city": "Bangalore",
              "email": "leanne.graham@gmail.com",
              "firstName": "Leanne",
              "gender": "Male",
              "id": 1001,
              "lastName": "Graham",
              "medicalConditions": Array [
                Object {
                  "id": 1,
                  "name": "Cephalalgia (headache)",
                },
                Object {
                  "id": 3,
                  "name": "Otitis externa (swimmer's ear)",
                },
                Object {
                  "id": 9,
                  "name": "Contusion (bruise)",
                },
              ],
              "phoneNumber": 9123456780,
              "state": "Karnataka",
            },
            "reaction": "Increased Perspiration",
            "severity": "MODERATE",
            "status": "ACTIVE",
          },
          Object {
            "encounter": Object {
              "date": "2021-10-26",
              "id": 1,
              "patient": Object {
                "age": 56,
                "city": "Mysore",
                "email": "clementine.bauch@gmail.com",
                "firstName": "Clementine",
                "gender": "Female",
                "id": 1003,
                "lastName": "Bauch",
                "medicalConditions": Array [
                  Object {
                    "id": 23,
                    "name": "Gestation (pregnancy) ",
                  },
                ],
                "phoneNumber": 9123456782,
                "state": "Karnataka",
              },
              "user": Object {
                "accountNonExpired": true,
                "accountNonLocked": true,
                "authorities": Array [
                  Object {
                    "authority": "USER",
                    "id": 1,
                    "roleCode": "USER",
                    "roleDescription": "User role",
                  },
                  Object {
                    "authority": "ADMIN",
                    "id": 2,
                    "roleCode": "ADMIN",
                    "roleDescription": "Admin role",
                  },
                ],
                "credentialsNonExpired": true,
                "email": "john.cage@gmail.com",
                "enabled": true,
                "id": 1,
                "password": "$2a$10$uS1MVfXzQkqbY7XDruby/OySJp3yjLmo4QGDSiE30GoN0wGb7DCqS",
                "role": "Physician",
                "username": "John",
              },
            },
            "id": 14,
            "ingredient": Object {
              "id": 3,
              "name": "Ibuprofen",
            },
            "name": "Paracetamol Allergy",
            "patient": Object {
              "age": 21,
              "city": "Bangalore",
              "email": "leanne.graham@gmail.com",
              "firstName": "Leanne",
              "gender": "Male",
              "id": 1001,
              "lastName": "Graham",
              "medicalConditions": Array [
                Object {
                  "id": 1,
                  "name": "Cephalalgia (headache)",
                },
                Object {
                  "id": 3,
                  "name": "Otitis externa (swimmer's ear)",
                },
                Object {
                  "id": 9,
                  "name": "Contusion (bruise)",
                },
              ],
              "phoneNumber": 9123456780,
              "state": "Karnataka",
            },
            "reaction": "Increased Perspiration",
            "severity": "MODERATE",
            "status": "ACTIVE",
          },
        ],
        "city": "Bangalore",
        "email": "leanne.graham@gmail.com",
        "firstName": "Leanne",
        "gender": "Male",
        "id": 1001,
        "lastName": "Graham",
        "medicalConditions": Array [
          Object {
            "id": 1,
            "name": "Cephalalgia (headache)",
          },
          Object {
            "id": 3,
            "name": "Otitis externa (swimmer's ear)",
          },
          Object {
            "id": 9,
            "name": "Contusion (bruise)",
          },
        ],
        "phoneNumber": 9123456780,
        "state": "Karnataka",
      }
    }
  >
    <Button
      active={false}
      data-testid="openAddAllergyModal"
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      <button
        className="btn btn-primary"
        data-testid="openAddAllergyModal"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Add Allergy
      </button>
    </Button>
    <Modal
      animation={true}
      aria-labelledby="contained-modal-title-vcenter"
      autoFocus={true}
      backdrop={true}
      centered={true}
      data-testid="addAllergyModal"
      dialogAs={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        }
      }
      enforceFocus={true}
      keyboard={true}
      onHide={[Function]}
      restoreFocus={true}
      show={false}
      size="md"
      style={
        Object {
          "zIndex": 2000,
        }
      }
    >
      <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={[Function]}
        containerClassName="modal-open"
        enforceFocus={true}
        keyboard={true}
        manager={
          BootstrapModalManager {
            "containers": Array [],
            "data": Array [],
            "handleContainerOverflow": true,
            "hideSiblingNodes": true,
            "modals": Array [],
            "scrollbarSize": 0,
          }
        }
        onEnter={[Function]}
        onEntering={[Function]}
        onEscapeKeyDown={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onHide={[Function]}
        renderBackdrop={[Function]}
        renderDialog={[Function]}
        restoreFocus={true}
        show={false}
        transition={[Function]}
      />
    </Modal>
  </AddAllergyPopUp>
</Router>
`;
